### PR TITLE
Replace request module with something more lightweight

### PR DESCRIPTION
**Pull request by christian-bromann**
## Problem

Currently the module uses request which comes with a lot of dependencies and is impossible to use in the browser (when bundled with e.g. webpack)

## Proposed Solution

Let's replace it with something more lightweight.

See also webdriverio/webdriverio#4235

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,17 +47,16 @@ If set to true you are accessing the headless Sauce instances (this discards the
 Type: `boolean`<br>
 Default: `false`
 
-## Usage
+### proxy
 
-All accessible API commands with descriptions can be found [here](docs/interface.md). In order to route requests through a proxy set `HTTP_PROXY`, `HTTPS_PROXY` or `NO_PROXY` as environment variable. For more information on this, read [here](https://github.com/request/request#controlling-proxy-behaviour-using-environment-variables). Alternatively you can pass the proxy to be used during initialization:
+If you want to tunnel your API request through a proxy please see the [got proxy docs](https://github.com/sindresorhus/got/blob/master/readme.md#proxies) for more information.
```
Comment (enriquegh) [inline README.md]: How would one add this to this package? Seems like it needs to be added when initializing got

Comment (enriquegh) [inline README.md]: Looking at got's README, you need to install either tunnel or global packages to be able to proxy the requests?

**Comment by enriquegh** [comment]
I was able to proxy the call through the command line doing something like:

`HTTP_PROXY=http://localhost:8889 ./bin/sl getCurrentUserFull`
but using -p/--proxy didn't seem to work at all.

Tried:
`HTTP_PROXY=http://localhost:8889 ./bin/sl getCurrentUserFull -p http://localhost:8889`
but based on the docs there might be more things that need to be done?

Should we make this easier for the user to do this?

Haven't tried using this as a Node package but I assume this will be a somewhat similar experience.
Will try that out tomorrow.

File: README.md (hunk 1 above)
Comment (christian-bromann) [reply README.md]: I am not entirely sure what the common use case for users are therefor I thought the best would be to do a major release and have them figure out what works for them. I am happy to make it simpler moving on.

**Comment by christian-bromann** [comment]
@enriquegh do you know the common usage pattern for proxying requests is? I am happy to make this simpler but I can't copy/recreate what the `request` lib has done so far for us.

**Comment by enriquegh** [comment]
@christian-bromann maybe @wswebcreation has a bit more insight being in the field but customer's (especially in enterprise environments) need all traffic proxied.

Like I mentioned, if they have the HTTP_PROXY environment variable set up they should be fine but I can't say if all/most of customer's would have this set up from the get go.

We could:
- Add a note saying that if you're using node-saucelabs as a CLI and need to proxy requests, use the HTTP_PROXY environment variable
- If they want to use this as an npm package, we can then either just link to got's docs or that and provide a very simple example on how to do it.

Would that be do-able?

**Comment by christian-bromann** [comment]
@enriquegh I will move ahead with this given that `request` has been officially deprecated. I will do a major release so customers won't be affected. Once we see customers struggle with it we can look into how to make setting proxies easier.

**Comment by enriquegh** [comment]
@christian-bromann 
I am good with figuring it out as we go but I would like to suggest to delete the proxy configuration part of the docs and instead mention that adding HTTP_PROXY as an environment variable should work.

Reason behind is because the proxy object is not working and will confuse people.

I can make a PR for the doc change (if needed) and an issue for fixing the proxy flag once I figure out what needs to be done.

**Comment by christian-bromann** [comment]
> to delete the proxy configuration part of the docs and instead mention that adding HTTP_PROXY as an environment variable should work.

Providing `HTTP_PROXY` as environment variable won't work anymore because Got doesn't support this.

> Reason behind is because the proxy object is not working and will confuse people.

Did you try this? I am passing the proxy object right into Got if it is not working it would be a bug on our side.

**Comment by enriquegh** [comment]
@christian-bromann 
Trying to test this but I can't run any command for some reason.
```
➜  node-saucelabs git:(master) ✗ nvm --version
0.33.11
➜  node-saucelabs git:(master) ✗ node --version
v12.13.0
```
Error:
```
➜  node-saucelabs git:(master) ✗ ./bin/sl getStatus

/Users/enriquegonzalez/git/node-saucelabs/src/index.js:78
            throw new Error(`Couldn't find API endpoint for command "${propName}"`)
                  ^
Error: Couldn't find API endpoint for command "then"
    at SauceLabs.get (/Users/enriquegonzalez/git/node-saucelabs/src/index.js:78:19)
    at isPromise (/Users/enriquegonzalez/git/node-saucelabs/node_modules/yargs/lib/is-promise.js:2:43)
    at Object.runCommand (/Users/enriquegonzalez/git/node-saucelabs/node_modules/yargs/lib/command.js:244:11)
    at Object.parseArgs [as _parseArgs] (/Users/enriquegonzalez/git/node-saucelabs/node_modules/yargs/yargs.js:1154:41)
    at Object.argv (/Users/enriquegonzalez/git/node-saucelabs/node_modules/yargs/yargs.js:1088:21)
    at Object.run (/Users/enriquegonzalez/git/node-saucelabs/src/cli.js:71:17)
    at Object.<anonymous> (/Users/enriquegonzalez/git/node-saucelabs/bin/sl:3:25)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
```
Tests are passing but I can't actually use it as a CLI.
Will keep trying but I make an issue with more info

**Comment by enriquegh** [comment]
Made:
https://github.com/saucelabs/node-saucelabs/issues/77
Let's talk there on fixing v4. Once we jump that hurdle I'll check the proxy stuff

**Pull request merged**